lib/tasks/javascript.rake

```diff
@@ -130,6 +130,135 @@ def absolute_sourcemap(dest)
   end
 end
 
+def generate_admin_sidebar_nav_map
+  vague_categories = { "root" => [] }
+
+  admin_routes =
+    Rails
+      .application
+      .routes
+      .routes
+      .map do |route|
+        next if route.verb != "GET"
+        path = route.path.spec.to_s.gsub("(.:format)", "")
+        next if !path.include?("admin")
+        next if path.include?("/:") || path.include?("admin-login")
+        path
+      end
+      .compact
+
+  engine_routes =
+    Rails::Engine
+      .subclasses
+      .map do |engine|
+        engine
+          .routes
+          .routes
+          .map do |route|
+            next if route.verb != "GET"
+            path = route.path.spec.to_s.gsub("(.:format)", "")
+            next if !path.include?("admin")
+            next if path.include?("/:") || path.include?("admin-login")
+            path
+          end
+          .compact
+      end
+      .flatten
+
+  admin_routes = admin_routes.concat(engine_routes)
+
+  admin_routes.each do |path|
+    split_path = path.split("/")
+    if split_path.length >= 3
+      vague_categories[split_path[2]] ||= []
+      vague_categories[split_path[2]] << { path: path }
+    else
+      vague_categories["root"] << { path: path }
+    end
+  end
+
+  # Copy this JS to your browser to get the Ember routes.
```

Comment:
I don't think there is a nicer way to do this, since I need a booted copy of ember, but I also have no idea. Happy to have a different way!